samples/copyOldVersionOfFile.js

```diff
@@ -53,12 +54,26 @@ function main(
 
   async function copyOldVersionOfFile() {
     // Copies the file to the other bucket
+
+    // Optional:
+    // Set a generation-match precondition to avoid potential race conditions
+    // and data corruptions. The request to upload is aborted if the object's
+    // generation number does not match your precondition. For a destination
+    // object that does not yet exist, set the ifGenerationMatch precondition to 0
+    // If the destination object already exists in your bucket, set instead a
+    // generation-match precondition using its generation number.
+    const copyOptions = {
+      preconditionOpts: {
+        ifGenerationMatch: generationMatchPrecondition,
```

Comment:
Probably should have called this out above but same comment for all the places that the precondition applies to the destination.

Reply:
updated this one and move. Good callout.